onnxruntime/python/tools/transformers/onnx_model.py

```diff
@@ -918,20 +937,37 @@ def prune_graph(self, outputs=None, allow_remove_graph_inputs=True):
             allow_remove_graph_inputs (bool): allow remove graph inputs.
         """
 
-        if len(self.graphs()) > 1:
-            # TODO(tianleiwu): handle subgraph
-            logger.debug("Skip prune_graph since graph has subgraph")
-            return
-
         keep_outputs = [output.name for output in self.model.graph.output] if outputs is None else outputs
 
+        input_name_to_nodes_for_main_graph = self.input_name_to_nodes_for_main_graph()
         output_name_to_node = self.output_name_to_node()
 
         def get_first_output(node):
             if node.output[0]:
                 return node.output[0]
             return next(iter([o for o in node.output if o]), None)
 
+        if len(self.graphs()) > 1:
+            # Get input names for all nodes in all subgraphs
+            subgraph_nodes = list(filter(lambda node: node.op_type in {"Loop", "Scan", "If"}, self.model.graph.node))
```

Comment:
other ops like BeamSearch also has subgraph. May add some TODO comments if we do not handle them right now.

Reply:
Added a TODO comment and kept the return early logic